License2Deploy/rolling_deploy.py

```diff
@@ -0,0 +1,60 @@
+import boto.ec2.autoscale as asg
+import re
+
+
+
+def return_autoscalinggroup_desired_size(auto_scaling_group):
+    """ AWS Call: return the  minsize of the AutoScaling Group of the Microservice"""
+    return auto_scaling_group.min_size
+
+def calculate_new_desired_ec2_count(current_desired_ec2_count):
+    """ By default, we double the instance Count """
+    return 2 * current_desired_ec2_count
+
+def update_desired_ec2_count_in_autoscalinggroup(auto_scaling_group, ec2_count):
+    auto_scaling_group.set_capacity(ec2_count)
+    """ AWS Call: update the desired_size of the AutoScaling Group of the Microserice"""
+    pass
+
+def healthcheck_new_instances():
+    """ AWS Call: Check whether the new instances are working good """
+    return True
+
+
+def update_ami_id_with_last_successfull():
+    """ AWS Call: As health check failed, update the ami_id with the last successfull one """
+    pass
+
+def retrieve_auto_scaling_group(app_name, environment):
+    """ AWS Call: retrieve the Auto Scaling Group by Tag """
+    auto_scaling_groups = asg.connect_to_region("us-east-1").get_all_groups();
+    return filter_auto_scaling_group(auto_scaling_groups, app_name, environment)
```

Comment:
When defining the filter_auto_scaling_group function, you are looking for four values (auto_scaling_groups, app_name, project_name, environment), but are only passing in three here: (auto_scaling_groups, app_name, environment)